styles.css

```diff
@@ -1,4 +1,28 @@
-.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder-title.file-color-file,
-.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-file-title.file-color-file {
+/* TEXT COLORING */
+/* These are applied when "inherit" is turned off -- only for individual items */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder-title.file-color-file.file-color-type-text,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-file-title.file-color-file.file-color-type-text,
+/* These are applied when "inherit" is turned on -- applies to the whole folder tree */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder.file-color-file.file-color-type-text .nav-folder-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder.file-color-file.file-color-type-text .nav-file-title,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-file.file-color-file.file-color-type-text .nav-file-title /* for top-level notes */
+{
   color: var(--file-color-color);
-}
\ No newline at end of file
+}
+
+/* BACKGROUND COLORING */
+/* These are applied when "inherit" is turned off -- only for individual items */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder-title.file-color-file.file-color-type-background,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-file-title.file-color-file.file-color-type-background,
+/* These are applied when "inherit" is turned on -- applies to the whole folder tree */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder.file-color-file.file-color-type-background,
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-file.file-color-file.file-color-type-background
+{
+  background-color: var(--file-color-color);
+}
+
+/* When inheritance turned on with background, each section looks better with rounded corners.
+   They are not rounded above with individual items because that appears to be styled by individual themes already, and we don't want to override them. */
+.workspace-leaf-content[data-type="file-explorer"] .nav-files-container .nav-folder.file-color-file.file-color-type-background {
+  border-radius: var(--radius-s);
+}
```

Comment:
I think treating a folder as a colored "section" is a bit of a personal aesthetic decision, which shouldn't be the default.
I would argue the default should be always applying the background to the title elements as shown in #3 .

Reply:
Hmm maybe. I can change it to only color the title els, or add a toggle option called "Color Whole Background Sections". Do you have a preference?

Reply:
In the interest of keeping the PR as small as possible, let's just stick to coloring the title elements for now. 🙂